components/dashboard/src/data/auth-providers/update-org-auth-provider-mutation.ts

```diff
@@ -0,0 +1,42 @@
+/**
+ * Copyright (c) 2023 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import { useMutation, useQueryClient } from "@tanstack/react-query";
+import { getOrgAuthProvidersQueryKey } from "./org-auth-providers-query";
+import { UpdateAuthProviderRequest } from "@gitpod/public-api/lib/gitpod/v1/authprovider_pb";
+import { authProviderClient } from "../../service/public-api";
+
+type UpdateAuthProviderArgs = {
+    provider: {
+        id: string;
+        clientId: string;
+        clientSecret: string;
+    };
+};
+export const useUpdateOrgAuthProviderMutation = () => {
```

Comment:
Thought about generalizing it, but exposing details about caches to calling client felt way more dirty than duplicating this. Also, we deliberately started with duplicating to get Git Integrations running for Orgs to avoid creating more spaghetti which which would need to be pulled apart when we remove the user-level integrations, which I hope we can start working on soon.

Reply:
That said, this duplication should make it easier to remove the user-level without risking to break anything.

Reply:
Then maybe a internal function, then export two function to call it?

Reply:
I prefer to leave it as is, because it allows to remove the user-level auth provider support easily.